pytorch_lightning/plugins/checkpoint/checkpoint.py

```diff
@@ -0,0 +1,22 @@
+from abc import ABC
+from pathlib import Path
+from typing import Any, Dict, Union
+
+
+class CheckpointIOPlugin(ABC):
+    def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: str) -> None:
+        """Save model/training states as a checkpoint file through state-dump and file-write.
+
+        Args:
+            checkpoint: dict containing model and trainer state
+            filepath: write-target file's path
+        """
+
+    def load_checkpoint_file(self, checkpoint_path: Union[str, Path]) -> Dict[str, Any]:
+        """
+        Load checkpoint from a path when resuming or loading ckpt for test/validate/predict stages.
+        Args:
+            checkpoint_path: Path to checkpoint
+
+        Returns: The loaded checkpoint.
+        """
```

Comment:
I think it could also be useful to support something like `storage_options` as a Dict passed in here
fsspec support this for their filesystem protocol: https://filesystem-spec.readthedocs.io/en/latest/_modules/fsspec/registry.html#filesystem

You could imagine someone wants to save a checkpoint with a specific TTL, for instance. This could be something that each Plugin implementation knows about, and which can be passed through the callback -> Trainer -> training type -> checkpoint plugin to support. 

@RahulMitra @yifuwang what do you think?

Reply:
It would come in handy if different invocations need to specify different storage options. API-wise this is something we can add as the need arises, but for FB's use case I think we already know we need it so you have my upvote :)

Reply:
Could this not be extended from the base class? Is this something that needs to happen in the function signature?